tests/python/helpers.py

```diff
@@ -1,45 +1,197 @@
+    """Asynchronously create accounts and return a transaction Builder instance for each account.
+
+    Each Builder instance uses a different Horizon endpoint if given a non-empty list.
+    """
+    accounts_to_create_num = len(channels) * MAX_OPS
+    logging.info('creating %d accounts', accounts_to_create_num)
+
+    accounts_to_create = [Keypair() for _ in range(accounts_to_create_num)]
+    xdrs = []
+    for i, channel in enumerate(channels):
+        for kp in accounts_to_create[i*MAX_OPS : i*(MAX_OPS) + MAX_OPS]:
+            # use the root account as the source of the operations
+            #
+            # NOTE channels need a starting balance since they need to pay fees because they are not prioritized
+            channel.append_create_account_op(source=root_account.address(),
+                                             destination=kp.public_address,
+                                             starting_balance=str(starting_balance))
+
+        # sign with channel and root account
+        channel.sign(secret=channel.keypair.seed().decode())
+        if root_account.seed().decode() != channel.keypair.seed().decode():
+            channel.sign(secret=root_account.seed().decode())
+        xdrs.append(channel.gen_xdr())
+
+        channel.next()
+
+    horizon_uri = channels[0].horizon.horizon_uri
+    await send_txs(horizon_uri, xdrs, expected_statuses=[200])
+
+    # fetch sequence numbers asynchronously for all created accounts
+    sequences = await get_sequences(horizon_uri, [a.public_address for a in accounts_to_create])
+
+    # create tx builders with up-to-date sequence number
+    builders = []
+    horizons = ([Horizon(horizon_uri=addr, user_agent=SDK_USER_AGENT) for addr in horizon_addresses]
+                if horizon_addresses
+                else [horizon_uri])
+    for i, kp in enumerate(accounts_to_create):
+        # deterministically pick horizon endpoint from all available endpoint
+        # according to keypair index in account list
+        horizon = horizons[i % len(horizons)]
+        builder = Builder(env.name, horizon, MIN_FEE, kp.secret_seed)
+        builder.sequence = sequences[i]
+        builders.append(builder)
+
+    for b in builders:
+        logging.debug('created builder %s', b.address)
+
+    return builders
+
+
+def init_tx_builders(env, kps, sequences):
+    """Initialize transaction builders for each given seed."""
+    builders = []
+    for i, kp in enumerate(kps):
+        client = kin.KinClient(env)
+        builder = Builder(env.name, client.horizon, MIN_FEE, kp.secret_seed)
+        builder.sequence = sequences[i]
+        builders.append(builder)
+    return builders
+
+
+async def get(session: aiohttp.ClientSession, url, expected_status=200):
+    """Send an HTTP GET request and return response JSON data.
+
+    Fail if response isn't expected status code or format other than JSON.
+    """
+    async with session.get(url) as res:
+        res_data = await res.json()
```

Comment:
```json(encoding='utf-8')``` , otherwise it will try to figure out the encoding itself, which is going to add some time if you do it tens of thousands of times

Reply:
yeah i did that at first but decided to keep it simple. this is far from being the computational bottleneck in this script